### PR TITLE
services/horizon: add option to submit without waiting for transaction final state

### PR DESCRIPTION
**Pull request by leighmcculloch**
<details>
  <summary>PR Checklist</summary>
  
### PR Structure

* [ ] This PR has reasonably narrow scope (if not, break it down into smaller PRs).
* [ ] This PR avoids mixing refactoring changes with feature changes (split into two PRs
  otherwise).
* [ ] This PR's title starts with name of package that is most changed in the PR, ex.
  `services/friendbot`, or `all` or `doc` if the changes are broad or impact many
  packages.

### Thoroughness

* [ ] This PR adds tests for the most critical parts of the new functionality or fixes.
* [ ] I've updated any docs ([developer docs](https://www.stellar.org/developers/reference/), `.md`
  files, etc... affected by this change). Take a look in the `docs` folder for a given service,
  like [this one](https://github.com/stellar/go/tree/master/services/horizon/internal/docs).

### Release planning

* [ ] I've updated the relevant CHANGELOG ([here](services/horizon/CHANGELOG.md) for Horizon) if
  needed with deprecations, added features, breaking changes, and DB schema changes.
* [ ] I've decided if this PR requires a new major/minor version according to
  [semver](https://semver.org/), or if it's mainly a patch change. The PR is targeted at the next
  release branch if it's not a patch change.
</details>

### What

Add option to submit without waiting for transaction final state. Transactions submitted with the additional parameter `async=true` are submitted directly to the submitter (stellar-core) and a pending transaction response is returned with a HTTP status code 202 Accepted.

### Why

Applications do not always want to keep a http connection open for ~5 seconds to find out the final state of the submitted transaction. In some cases applications would prefer to submit and lookup the transaction later to see if it was successful, failed, or just not included.

Technically if an application was to cover all edge cases it would need to support that later lookup anyway, because it is possible for the Horizon to timeout, or the connection to be interrupted. It's also always possible for a submitted transaction to be scooped up by some other party if the transaction has been broadcasted or shared, and for that other party to resubmit the transaction during its time bound window of validity, even if Horizon fails the transaction immediately on the initial submit.

For applications that implement these edge cases already, submitting without waiting may be attractive.

### Known limitations

This PR includes no tests for the new behavior of the endpoint. I've been working on a test, but it's a work-in-progress, and I'm learning a bit about how action tests work so it's taking me a bit longer.

### CODE REVIEW

**Comment by leighmcculloch** [comment]
@bartekn I'd like to make this change an experimental feature of Horizon that is only enabled if explicitly indicated. Is there a common way to flag and enable/disable experimental features in Horizon? Or should I just add an new config option to the top-level CLI?

**Review by Shaptic — COMMENTED** [review]
This seems like a very valuable addition to our txsub model! I know I wasn't requested for review on this, but I thought I'd drop some thoughts anyway, mostly to clarify my own understanding of intent here.

File: services/horizon/internal/actions/helpers.go [hunk 1]
```diff
@@ -146,6 +146,28 @@ func getString(r *http.Request, name string) (string, error) {
...
+	}
+
+	asBool, err := strconv.ParseBool(str)
+	if err != nil {
+		return false, problem.MakeInvalidFieldProblem(name, errors.New("unparseable value"))
+	}
+
+	return asBool, nil
+}
+
```
[the remark this answers is not in the document]
Comment (leighmcculloch) [reply services/horizon/internal/actions/helpers.go:170]: I could only find a GetString function. It's possible I missed it. Thanks for pointing it out. This could be replaced with any existing function if there is one.

**Comment by bartekn** [comment]
GH autoclosed this because I removed an old release branch. Unfortunately I can't reopen. Can you reopen against `master` this time?

File: services/horizon/internal/actions/helpers.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Shaptic) [reply services/horizon/internal/actions/helpers.go:170]: Ah, we use a library for this I guess (see `helpers.go::getParams::403`) which is catered towards HTTP-style URLs.

**Comment by leighmcculloch** [comment]
@bartekn GitHub does not allow me to reopen the PR, or change its base branch. I believe someone else may be taking on turning this feature request and spike into a full fledged feature. If that happens and they'd like to use this code, they can checkout the code from this PR.
